android/app/src/main/java/com/microsoft/codepush/react/CodePushUpdateManager.java

```diff
@@ -223,67 +223,70 @@ public void downloadPackage(JSONObject updatePackage, String expectedBundleFileN
             String diffManifestFilePath = CodePushUtils.appendPathComponent(unzippedFolderPath,
                     CodePushConstants.DIFF_MANIFEST_FILE_NAME);
             boolean isDiffUpdate = FileUtils.fileAtPathExists(diffManifestFilePath);
+
             if (isDiffUpdate) {
                 String currentPackageFolderPath = getCurrentPackageFolderPath();
                 CodePushUpdateUtils.copyNecessaryFilesFromCurrentPackage(diffManifestFilePath, currentPackageFolderPath, newUpdateFolderPath);
                 File diffManifestFile = new File(diffManifestFilePath);
                 diffManifestFile.delete();
             }
 
-            FileUtils.copyDirectoryContents(unzippedFolderPath, newUpdateFolderPath);
-            FileUtils.deleteFileAtPathSilently(unzippedFolderPath);
-
             // For zip updates, we need to find the relative path to the jsBundle and save it in the
             // metadata so that we can find and run it easily the next time.
-            String relativeBundlePath = CodePushUpdateUtils.findJSBundleInUpdateContents(newUpdateFolderPath, expectedBundleFileName);
-
+            String relativeBundlePath = CodePushUpdateUtils.findJSBundleInUpdateContents(unzippedFolderPath, expectedBundleFileName);
             if (relativeBundlePath == null) {
-                throw new CodePushInvalidUpdateException("Update is invalid - A JS bundle file named \"" + expectedBundleFileName + "\" could not be found within the downloaded contents. Please check that you are releasing your CodePush updates using the exact same JS bundle file name that was shipped with your app's binary.");
-            } else {
-                if (FileUtils.fileAtPathExists(newUpdateMetadataPath)) {
-                    File metadataFileFromOldUpdate = new File(newUpdateMetadataPath);
-                    metadataFileFromOldUpdate.delete();
+                FileUtils.deleteFileAtPathSilently(unzippedFolderPath);
+                if (FileUtils.fileAtPathExists(newUpdateFolderPath)) {
+                    FileUtils.deleteFileAtPathSilently(newUpdateFolderPath);
                 }
+                throw new CodePushInvalidUpdateException("Update is invalid - A JS bundle file named \"" + expectedBundleFileName + "\" could not be found within the downloaded contents. Please check that you are releasing your CodePush updates using the exact same JS bundle file name that was shipped with your app's binary.");
+            }
 
-                if (isDiffUpdate) {
-                    CodePushUtils.log("Applying diff update.");
-                } else {
-                    CodePushUtils.log("Applying full update.");
-                }
+            FileUtils.copyDirectoryContents(unzippedFolderPath, newUpdateFolderPath);
```

Comment:
Could you please add removing of unzipped folder like in Master? 
`FileUtils.deleteFileAtPathSilently(unzippedFolderPath);`
From here: https://github.com/Microsoft/react-native-code-push/blob/master/android/app/src/main/java/com/microsoft/codepush/react/CodePushUpdateManager.java#L234

Reply:
@alexandergoncharov Please notice https://github.com/Microsoft/react-native-code-push/pull/1367/files/1378ed54ab5c73dafcd2a49d17974db26515b357#diff-4aa56c56a1afb536396acce24d2d2015R242 (CodePushInvalidUpdateException  will be thrown & by adding the line as in master would leave the copy of unzipped folder in the memory as the statement will not be executed, so the PR moves the delete operation before the exception is thrown)

Reply:
@krunalsshah , yeah, That's great! You made the good thing that you've added removing unzippedFolderPath and newUpdateFolderPath before Error. But in the successful case(without any Error) unzipped folder will not be deleted at all. But it should be deleted after coping context.

Please correct me if I missed something.

Reply:
@alexandergoncharov yes you are absolutely correct. I have pushed the commit.

Reply:
@krunalsshah Thanks!